test/tests/functional/pbs_basil_support.py

```diff
@@ -465,28 +461,24 @@ def test_job_request_knl(self):
         job = Job(TEST_USER)
 
         job.create_script(
-                          "#PBS -joe -o localhost:/tmp -lselect=1:ncpus=1:aoe=%s\n" % req_aoe +
-                          "cd /tmp\n" +
-                          "aprun -B sleep 10\n" +
-                          "sleep 10")
+            "#PBS -joe -o localhost:/tmp -lselect=1:ncpus=1:aoe=%s\n"
+            % req_aoe +
+            " cd /tmp\n"
+            "aprun -B sleep 10\n"
+            "sleep 10")
 
         job_id = self.server.submit(job)
         self.server.expect(JOB, {'job_state': 'R'}, id=job_id)
 
         # Check that exec_vnode is a KNL vnode.
         self.server.status(JOB, 'exec_vnode', id=job_id)
-        evnode = job.execvnode()[0].keys()[0]
+        evnode = job.get_vnodes()[0]
         if evnode in klist:
             self.logger.info("exec_vnode %s is a KNL vnode." % (evnode))
-            rv = 1
+            rv = True
         else:
             self.logger.info("exec_vnode %s is not a KNL vnode." % (evnode))
-            rv = 0
-        self.assertTrue(rv == 1)
-
-        # Delete the Job.
-        self.server.delete(job_id, wait=True)
-
+        self.assertTrue(rv)
```

Comment:
with rv not being set to False on the else portion, won't this raise a ValueError then? (for the rest of these too).